src/autoreplace_gui.cpp

```diff
@@ -173,6 +174,10 @@ class ReplaceVehicleWindow : public Window {
 				if (num_engines == 0 && EngineReplacementForCompany(Company::Get(_local_company), eid, this->sel_group) == INVALID_ENGINE) continue;
 			} else {
 				if (!CheckAutoreplaceValidity(this->sel_engine[0], eid, _local_company)) continue;
+				if (e->IsVariantHidden(_local_company)) {
```

Comment:
Why move this here instead of keeping on L151? Assuming we want to `continue` as soon as possible for performance.

Reply:
IsVariantHidden needs to be the last check for num_hidden_engines to only count vehicles that were hidden and not removed from the list for other reasons.